RamenPos/MainForm.cs

```diff
@@ -210,18 +280,46 @@ private void Start()
     MessageBoxButtons.OK);
                 grpAutoAddressResolution.Enabled = false;
             }
+
+            if (!IsStarted)
+            {
+                this.Invoke(new MethodInvoker(delegate ()
+                {
+                    SpiStatusAndActions();
+                }));
+            }
         }
 
         private void OnDeviceAddressChanged(object sender, DeviceAddressStatus e)
         {
             this.Invoke(new MethodInvoker(delegate ()
             {
                 btnMain.Enabled = false;
-                if (!string.IsNullOrWhiteSpace(e?.Address))
+                if (e != null)
                 {
-                    txtAddress.Text = e.Address;
-                    btnMain.Enabled = true;
-                    MessageBox.Show($@"Device Address has been updated to {e.Address}", @"Device Address Updated");
+                    switch (e.ResponseCode)
+                    {
+                        case ResponseCode.SUCCESS:
+                            txtAddress.Text = e.Address;
+                            btnMain.Enabled = true;
+                            MessageBox.Show($@"Device Address has been updated to {e.Address}", @"Device Address Updated");
+                            break;
+                        case ResponseCode.ERROR:
+                            txtAddress.Text = "";
+                            MessageBox.Show("The serial number is invalid!");
```

Comment:
This is probably not true, if the device service is down it will return an error as well, you will need to update SendRequest<T> to return the correct response